api/v1alpha1/progressiverollout_types.go

```diff
@@ -28,18 +30,68 @@ type ProgressiveRolloutSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of ProgressiveRollout. Edit ProgressiveRollout_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// SourceRef references the resource, example an ApplicationSet, which owns ArgoCD Applications
+	//+kubebuilder:validation:Required
+	SourceRef corev1.TypedLocalObjectReference `json:"sourceRef"`
+	// Stages reference a list of Progressive Rollout stages
+	//+kubebuilder:validation:Optional
+	Stages []*ProgressiveRolloutStage `json:"stages,omitempty"`
```

Comment:
Why is this a reference?

Reply:
I'm guessing it's because the subsequent methods for ProgressiveRolloutStage are ref-based as they will change over time.
@maruina is this your understanding as well?

Reply:
No I think it's a good point, I don't think it has to be a reference. This object will store the information and it's read-only. I'll fix that.